x/incentives/keeper/evm_hooks.go

```diff
@@ -41,7 +41,7 @@ func (h Hooks) PostTxProcessing(ctx sdk.Context, msg core.Message, receipt *etht
 	}
 
 	ethAccount, ok := acc.(ethermint.EthAccountI)
-	if !ok || ethAccount.Type() != ethermint.AccountTypeEOA {
+	if ok && ethAccount.Type() == ethermint.AccountTypeContract {
```

Comment:
@fedekunze @ramacarlucho I think the check `ethAccount.Type() == ethermint.AccountTypeContract` doesn't work as expected. I ran this through a debugger and `ethAccount.Type()` equals `2` even though I made a tx from an EOA.